packages/suite/src/utils/suite/logsUtils.ts

```diff
@@ -102,34 +128,63 @@ export const redactAction = (action: Action): Action => {
     };
 };
 
-export const redactCustom = (action: CustomLogEntry['action']) => {
-    let payload;
-    switch (action.type) {
-        case SUITE.AUTH_DEVICE:
-            payload = {
-                state: REDACTED_REPLACEMENT,
-                payload: redactDevice(action.payload),
-            };
-            break;
-        case TRANSACTION.ADD:
-            payload = {
-                ...action.payload,
-                deviceState: REDACTED_REPLACEMENT,
-                descriptor: REDACTED_REPLACEMENT,
-            };
-            break;
-        case ACCOUNT.UPDATE_SELECTED_ACCOUNT:
-            payload = {
-                ...action.payload,
-                account: redactAccount(action.payload?.account),
-            };
-            break;
+export const getApplicationLog = (log: LogEntry[], redactSensitiveData = false) =>
+    log.map(entry => {
+        const metadata = {
+            type: entry.type,
+            datetime: entry.datetime,
+        };
 
-        default:
-            return action;
-    }
-    return {
-        ...action,
-        payload,
-    };
-};
+        let redactedAction = entry.payload;
+        if (redactSensitiveData) {
+            redactedAction = redactAction(entry);
+        }
+
+        if (typeof redactedAction?.payload === 'object') {
+            return { ...metadata, payload: { ...redactedAction.payload } };
+        }
+
+        return {
+            ...metadata,
+            payload: {
+                ...redactedAction,
+            },
+        };
+    });
+
+export const getApplicationInfo = (state: AppState) => ({
+    environment: getEnvironment(),
+    suiteVersion: process.env.VERSION || '',
+    commitHash: process.env.COMMITHASH || '',
+    isDev: !process.env.CODESIGN_BUILD,
+    debugMenu: state.suite.settings.debug.showDebugMenu,
+    online: state.suite.online,
+    browserName: getBrowserName(),
+    browserVersion: getBrowserVersion(),
+    osName: getOsName(),
+    osVersion: getOsVersion(),
+    windowWidth: getWindowWidth(),
+    windowHeight: getWindowHeight(),
+    screenWidth: getScreenWidth(),
+    screenHeight: getScreenHeight(),
+    earlyAccessProgram: state.desktopUpdate.allowPrerelease,
+    language: state.suite.settings.language,
+    autodetectLanguage: state.suite.settings.autodetect.language,
+    platformLanguages: getPlatformLanguages().join(','),
+    theme: state.suite.settings.theme.variant,
+    autodetectTheme: state.suite.settings.autodetect.theme,
+    localCurrency: state.wallet.settings.localCurrency,
+    discreetMode: state.wallet.settings.discreetMode,
+    tor: getIsTorEnabled(state.suite.torStatus),
+    torOnionLinks: state.suite.settings.torOnionLinks,
+    labeling: state.metadata.enabled ? state.metadata.provider?.type || '' : '',
```

Comment:
what about to distinguish the state when metadata are enabled, but there is no provider?
```suggestion
    labeling: state.metadata.enabled ? state.metadata.provider?.type || 'missing-provider' : '',
```

Reply:
Done, squashed and rebased. I will add it also to analytics in separate PR